### PR TITLE
Generate and publish API stub file for Python

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Generate api stub file and publish it as artifact in package folder

### CODE REVIEW

File: scripts/devops_tasks/tox_harness.py [hunk 1]
```diff
@@ -435,6 +435,11 @@ def prep_and_run_tox(targeted_packages, parsed_args, options_array=[]):
         if parsed_args.tenvparallel:
             tox_execution_array.extend(["-p", "all"])
 
+        if parsed_args.tox_env == "apistub":
```
Comment (praveenkuttappan) [inline scripts/devops_tasks/tox_harness.py:438]: I prefer not to do this if tox has an option to choose a specific arg from a list of args in {posargs} and pass to it to command. Something like {posargs:out_path} but this didn't work.

File: eng/pipelines/templates/steps/analyze.yml [hunk 2]
```diff
@@ -108,12 +108,6 @@ steps:
       artifactName: 'packages'
       targetPath: $(Build.ArtifactStagingDirectory)
 
-  - template: ../steps/run_apistub.yml
-    parameters:
-      ServiceDirectory: ${{ parameters.ServiceDirectory }}
-      TestMarkArgument: ${{ parameters.TestMarkArgument }}
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:114]: Do we need any of these other args in the new usages?

Comment (praveenkuttappan) [reply eng/pipelines/templates/steps/analyze.yml:114]: We have removed this step in analyze and only service directory is required in build job. I will remove other params from template itself.

Comment (praveenkuttappan) [reply eng/pipelines/templates/steps/analyze.yml:114]: Done.

**Comment by praveenkuttappan** [comment]
@scbedd Can you please have a look at this PR when you get a chance?

File: scripts/devops_tasks/tox_harness.py (hunk 1 above)
Comment (scbedd) [reply scripts/devops_tasks/tox_harness.py:438]: I think this is fine. I see your difficulty.

**Review by scbedd — APPROVED** [review]

File: eng/tox/run_apistubgen.py [hunk 3]
```diff
@@ -56,4 +63,9 @@ def get_package_wheel_path(pkg_root):
     if not pkg_path:
         pkg_path = args.target_package
 
-    check_call(["apistubgen", "--pkg-path", pkg_path,], cwd=args.work_dir)
+    cmds = ["apistubgen", "--pkg-path", pkg_path]
```
Comment (weshaggard) [inline eng/tox/run_apistubgen.py:66]: For my understanding where do we install the apistubgen code from azure-sdk-tools repo?

**Pull request merged**